include/mbedtls/constant_time.h

```diff
@@ -9,6 +9,10 @@
 #ifndef MBEDTLS_CONSTANT_TIME_H
 #define MBEDTLS_CONSTANT_TIME_H
 
```

Comment:
I consider this a bug in 3.28 and 3.6, so I would prefer to backport it. However, we maintain ABI compatibility in LTS branches. We've never explicitly stated if that's just about the ABI itself, or also encompasses the ABI as seen through C++ linkage. Would adding `extern "C"` be considered an ABI break? I think not, since it doesn't change the generated binary, but maybe I'm missing something? What about a library that was written in C++ and built against Mbed TLS 3.6.0: would this patch break it if it's applied to 3.6.1?

Reply:
No, this does not affect the ABI at all, since the function definition itself is still written in C and compiled as such (i.e. not compiled by g++). The only problem was that the C++ declaration was wrong, causing a linker error.

I noticed this myself in a platformio project where I included this library (as a git submodule, not the outdated package), and found a strange link error, where only mbedtls_ct_memcmp wasn't found.

Reply:
Thanks for the information! Then please make backports to the `mbedtls-3.6` and `mbedtls-2.28` long-time support branches.